kyverno-policy-deployment/README.md

```diff
@@ -0,0 +1,51 @@
+# Handling Kyverno Policy Deployment
+
+Our Kyverno policies have the following properties:
+1. They are owned by different teams
+2. The need to be deployed to different environments
+3. They required different testing setups according to 1. and 2.
+
+Up until now we kept these policies in the same repository called `kyverno-policies`.
+Now that we will have to rename the repository to upgrade Kyverno is an opportune time to discuss the future structure of our Kyverno policies.
+
+## Proposals
+
+## Keep all policies in the same repository
+It's simple, everything stays the same and `kyverno-policies` is just renamed.
+### Pros
+- Not a lot of effort
+- Not much communication to make a change
+- Single source of truth for all policies
+### Cons
+- Testing setup is complex as all different environments need to be covered in one repo
```

Comment:
Can you give an example of this in the current state?

Reply:
The current setup is quite different to all our other projects. It leveraged a Python test framework and at times is a struggle to get running. For example, I personally haven't managed to successfully get the tests running on my machine so rely on CI on PRs to check if my changes work or not.

We also don't currently make use of the kyverno CLIs `test` command (mainly because at the time it wasn't able to cover most functionality)

Reply:
The python tests are using `ats`, which I agree is still difficult to use, but is theoretically the intended direction for app testing and is completely independent of the policies. That's actually a great example of why I'm hesitant about test quality. Multiple teams would have to duplicate and manage tests they aren't comfortable with. In the current setup, you benefit from existing tests even if you didn't write them.

I also agree it would be nice to use kyverno's own test capabilities going forward. If policies are managed in many places, it will be more difficult to align testing between them and will cause duplicated effort between teams.